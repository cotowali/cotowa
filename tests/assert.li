var a = 0
var b = 1
assert a == a
assert a != b
assert a == b
