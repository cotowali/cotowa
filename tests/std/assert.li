require 'assert'

module string {
  #[test]
  fn test_eq() {
    assert::string::eq('a', 'a')
    assert::string::eq('a', 'b')
  }

  #[test]
  fn test_ne() {
    assert::string::ne('a', 'b')
    assert::string::ne('a', 'a')
  }
}

module int {
  #[test]
  fn test_eq() {
    assert::int::eq(1, 1)
    assert::int::eq(1, 2)
  }

  #[test]
  fn test_ne() {
    assert::int::ne(1, 2)
    assert::int::ne(1, 1)
  }

  #[test]
  fn test_gt() {
    assert::int::gt(2, 1)
    assert::int::gt(1, 1)
  }

  #[test]
  fn test_ge() {
    assert::int::ge(2, 1)
    assert::int::ge(1, 1)
    assert::int::ge(0, 1)
  }

  #[test]
  fn test_lt() {
    assert::int::lt(1, 2)
    assert::int::lt(1, 1)
  }

  #[test]
  fn test_le() {
    assert::int::le(1, 2)
    assert::int::le(1, 1)
    assert::int::le(1, 0)
  }
}
