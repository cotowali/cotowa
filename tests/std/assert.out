testing string::test_eq ... FAILED

Output:
LINE 24: Assertion Failed: (left == right)
left : `a`
right: `b`

testing string::test_ne ... FAILED

Output:
LINE 28: Assertion Failed: (left != right)
left : `a`
right: `a`

testing int::test_eq ... FAILED

Output:
LINE 34: Assertion Failed: (left == right)
left : `1`
right: `2`

testing int::test_ne ... FAILED

Output:
LINE 38: Assertion Failed: (left != right)
left : `1`
right: `1`

testing int::test_gt ... FAILED

Output:
LINE 42: Assertion Failed: (left > right)
left : `1`
right: `1`

testing int::test_ge ... FAILED

Output:
LINE 46: Assertion Failed: (left >= right)
left : `0`
right: `1`

testing int::test_lt ... FAILED

Output:
LINE 50: Assertion Failed: (left < right)
left : `1`
right: `1`

testing int::test_le ... FAILED

Output:
LINE 54: Assertion Failed: (left <= right)
left : `1`
right: `0`

test result: FAILED
8 total; 0 passed; 8 failed;
